server/container/guice/memory-guice/src/main/java/org/apache/james/modules/mailbox/MemoryMailboxModule.java

```diff
@@ -78,6 +80,7 @@ protected void configure() {
         bind(MailboxManager.class).to(InMemoryMailboxManager.class);
         bind(MailboxACLResolver.class).to(UnionMailboxACLResolver.class);
         bind(GroupMembershipResolver.class).to(SimpleGroupMembershipResolver.class);
+        bind(new TypeLiteral<SystemMailboxesProvider<InMemoryId>>(){}).to(new TypeLiteral<SystemMailboxesProviderImpl<InMemoryId>>(){});
```

Comment:
Ideally yes.. But as there is no Cassandra/InMemory guice module in JMAP, I have to put registration in here I guess.